mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java

```diff
@@ -245,6 +237,69 @@ public UpdateControl<ServiceDomain> reconcile(ServiceDomain sd, Context context)
         }
     }
 
+    private Condition createConfigMaps(ServiceDomain sd, String sdTypeAsString, ApiVersion apiVersion, ServiceDomainSpec.Type sdType, String directVersion, String openApiVersion, String directConfigMapName) throws IOException {
+        final String openApiConfigMapName = sdTypeAsString + OPENAPI_CM_SUFFIX + "-" + openApiVersion;
+        String sdName = sd.getMetadata().getName();
+
+        final ConfigMap sdOpenApiConfigMap = client.configMaps().inNamespace(sd.getMetadata().getNamespace()).withName(openApiConfigMapName).get();
+        if (sdOpenApiConfigMap == null) {
+            final String openApiFileDir = "/" + apiVersion.name() + "/" + sdTypeAsString + "/";
+            final String openApiFileName = sdType + "-" + openApiVersion + ".json";
+
+            final InputStream openApiFileAsStream = getClass().getResourceAsStream(openApiFileDir + openApiFileName);
+            if (openApiFileAsStream == null) {
+                LOGGER.error("{} service domain cant read openapi mapping config file", sdName);
+                return new ConditionBuilder()
+                        .withType(CONDITION_INTEGRATION_READY)
+                        .withStatus(STATUS_FALSE)
+                        .withReason(REASON_CANT_READ_CONFIG_MAPS_FILE)
+                        .withMessage(MESSAGE_CANT_READ_CONFIG_MAPS_FILE)
+                        .build();
+            }
+
+            final ConfigMap openApiCM = new ConfigMapBuilder()
+                    .withNewMetadata()
+                    .withName(openApiConfigMapName)
+                    .withNamespace(sd.getMetadata().getNamespace())
+                    .withLabels(Map.of(MANAGED_BY_LABEL, OPERATOR_NAME))
+                    .endMetadata()
+                    .withData(Map.of(openApiFileName, IOUtils.toString(openApiFileAsStream, StandardCharsets.UTF_8)))
+                    .build();
+
+            client.configMaps().inNamespace(sd.getMetadata().getNamespace()).withName(openApiConfigMapName).create(openApiCM);
+        }
+
+        ConfigMap sdDirectConfigMap = client.configMaps().inNamespace(sd.getMetadata().getNamespace()).withName(directConfigMapName).get();
+        if (sdDirectConfigMap == null) {
+            final String directFileDir = "/" + apiVersion.name() + "/" + sdTypeAsString + "/";
```

Comment:
define a constant template for these 2 variables and just apply the format based on the variables